src/intelli/gui/graphscene.cpp

```diff
@@ -1423,20 +1321,143 @@ GraphScene::groupNodes(QVector<NodeGraphicsObject*> const& selectedNodeObjects)
     // make output connections
     index = PortIndex{0};
     type  = PortType::Out;
-    for (ConnectionId conId : qAsConst(connectionsOut))
+    for (ConnectionUuid const& conId : qAsConst(connectionsOut))
     {
         makeConnections(conId, outputProvider, index, type);
 
-        makeSharedConnetions(connectionsOutShared, conId, inputProvider, index, type);
+        makeSharedConnetions(connectionsOutShared, conId, outputProvider, index, type);
 
         index++;
     }
 
-    // make internal connections
-    for (ConnectionId const& conId : qAsConst(connectionsInternal))
+    restoreCmd.clear();
+}
+
+void
+GraphScene::expandGroupNode(Graph* groupNode)
+{
+    assert(groupNode);
+
+    // create undo command
+    auto appCmd = gtApp->makeCommand(m_graph, tr("Expand group node '%1'").arg(groupNode->caption()));
+    auto modifyGroupCmd = groupNode->modify();
+    auto modifyCmd = m_graph->modify();
+    Q_UNUSED(modifyCmd);
+
+    auto restoreCmd = gt::finally([&appCmd](){
+        appCmd.finalize();
+        gtApp->undoStack()->undo();
+    });
+
+    auto const& conModel = m_graph->connectionModel();
+
+    auto* inputProvider  = groupNode->inputProvider();
+    auto* outputProvider = groupNode->outputProvider();
+    assert(inputProvider);
+    assert(outputProvider);
+
+    // gather input and output connections
+    QVector<ConnectionUuid> expandedInputConnections, expandedOutputConnections;
+
+    {
+        auto const& groupConModel = groupNode->connectionModel();
+
+        auto inputCons  = groupConModel.iterateConnections(inputProvider->id(), PortType::Out);
+        for (ConnectionId conId : inputCons)
+        {
+            ConnectionUuid conUuid = groupNode->connectionUuid(conId);
+            auto connections = conModel.iterate(groupNode->id(), conUuid.outPort);
+            if (connections.empty()) continue;
+
+            for (auto const& connection : connections)
+            {
+                NodeId sourceNodeId = connection.node;
+                Node* sourceNode = m_graph->findNode(sourceNodeId);
+                assert(sourceNode);
+                gtDebug() << sourceNode->caption() << "TO" << groupNode->findNode(conId.inNodeId)->caption();
+                conUuid.outNodeId = sourceNode->uuid();
+                conUuid.outPort   = connection.port;
+
+                expandedInputConnections.push_back(conUuid);
+            }
+        }
+
+        gtDebug() << "SWITCH";
```

Comment:
Please give a more meaningful message here.

Reply:
thanks, this must be removed